src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -395,6 +395,12 @@ private void SetAnalyzedNullability(BoundExpression? expr, VisitResult result, b
         /// </summary>
         private int _lastConditionalAccessSlot = -1;
 
+        private BoundNode? _visitingConversionOperand;
+
+#if DEBUG
+        private readonly HashSet<BoundLambda> _errorLambdas = new HashSet<BoundLambda>();
```

Comment:
Consider leaving a comment such as "Lambdas can be visited with or without conversion, but the visitor should follow such and such rules"

Reply:
I described the rules inside `VisitLambda` and left only a short comment here.